### ISSUE TITLE
Unexpected error code 13 while fetching data

### ISSUE THREAD

**Issue by ad3luc** [Azure/azure-event-hubs-for-kafka#31]
Description
===========
I'm developing a SpringBoot app dockerized. The docker images are microservices and one of these communicate with Azure Event Hub.

Some of my properties:

**spring-boot -> 2.0.7.RELEASE**

**spring-cloud.version -> Finchley.SR2**

I've created a topic in Azure(with Kafka enabled).

I've follow some simple guide to set up my microservice and everything works fine.

```
@EnableBinding({Processor.class})
public class EventService {
    ...
    @Autowired private Processor ehProcessor;
    ...
    public void send(String event) {

        Message<String> message = MessageBuilder
                .withPayload(event)
                .setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.APPLICATION_JSON)
                .build();

        boolean send = ehProcessor.output().send(message, 5000L);

        if (!send) {

            log.error("Event NOT sent", event);
        }
    }

    ...

    @StreamListener(target = Processor.INPUT)
    public void receive(String event) {

        handle(event);
    }
}
```
For an entire month everything goes fine but in the last two days the microservice stucks because a continuous stacktrace is filling all my disc (a solution was to set up a docker log rotation).

```
java.lang.IllegalStateException: Unexpected error code 13 while fetching data
        at org.apache.kafka.clients.consumer.internals.Fetcher.parseCompletedFetch(Fetcher.java:891) ~[kafka-clients-1.0.1.jar!/:na]
        at org.apache.kafka.clients.consumer.internals.Fetcher.fetchedRecords(Fetcher.java:528) ~[kafka-clients-1.0.1.jar!/:na]
        at org.apache.kafka.clients.consumer.KafkaConsumer.pollOnce(KafkaConsumer.java:1154) ~[kafka-clients-1.0.1.jar!/:na]
        at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1111) ~[kafka-clients-1.0.1.jar!/:na]
        at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.run(KafkaMessageListenerContainer.java:699) ~[spring-kafka-2.1.7.RELEASE.jar!/:2.1.7.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_181]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_181]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_181]
```
I'm talking about 8/9 log messsage in 1 ms.

Looking inside the org.apache.kafka.common.protocol.Errors class the error is related to:

> NETWORK_EXCEPTION(13, "The server disconnected before a response was received.")

I'm not able to reproduce programmatically this error. I don't understand why when the first error is arised the log will start and no stop in infinite loop. I need to stop the docker container and sometimes the container will not stop. The only solution is to remove the container and recreate again.

How to reproduce
================
I'm not able to reproduce programmatically. It's happens every day after 18th on December

Has it worked previously?
=========================
At the beginning I was able to run the application without problem

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:
 
 - [x] Verified that port 9093 is not blocked by firewall
 - [x] Sample you're having trouble with: `<Java>`
 - [x] Spring cloud version: `<Finchley.SR2>`
 - [x] Kafka client configuration: `<only the auth properties to connect to the azure topic>` 
 - [x] Provide all logs (with debugging enabled if possible) or exception call stack
 - [x] Operating system: `<Ubuntu 18.04 (x64) LTS>`

**Comment by arerlend** [comment]
Can you email me your namespace? arerlend@microsoft.com

**Comment by arerlend** [comment]
Are you still seeing this error?

**Comment by ad3luc** [comment]
Sorry for the delay of my response.
The microservices is not running because the office is now closed. On
Monday I will restart it again.
I will provide you all the info that you need to investigate, so if you
find something strange we can close the issue on github.
If the error happens again we will see the Tuesday morning it usually
happens in early morning.

Have you already done some fix?

Best Regards,

Andrea De Luca

**Comment by arerlend** [comment]
Yes - Spring Boot gets stuck running a very tight retry loop on the same memory location when it fails to parse.  When we changed an UnknownServerException to a NetworkException, Spring Boot started getting stuck in the retry loop.  We reverted that exception type change and deployed the fix to all namespaces.

**Comment by ad3luc** [comment]
So I need to change some version in my pom?

Andrea De Luca

**Comment by arerlend** [comment]
Nope, the change was made service-side to bypass the retry loop problem, so you don't have to do anything.  Spring Boot is using an old version of the Kafka client w/ a bug (KAFKA-6877) - that bug was fixed in Kafka client 2.0.  It's up to Pivotal to update their dependencies.

**Comment by ad3luc** [comment]
Ok thanks!
I hope that I can give you more feedback as soon as possible.

Andrea De Luca

**Comment by arerlend** [comment]
All namespaces have been had the hotfix deployed.  @ad3luc confirmed no more errors, issue is closed.
